substrate/frame/recovery/src/lib.rs

```diff
@@ -235,9 +240,15 @@ pub mod pallet {
 			+ From<frame_system::Call<Self>>;
 
 		/// The currency mechanism.
-		type Currency: ReservableCurrency<Self::AccountId>;
+		#[cfg(feature = "runtime-benchmarks")]
+		type Currency: Mutate<Self::AccountId>
+			+ MutateHold<Self::AccountId, Reason = Self::RuntimeHoldReason>;
```

Comment:
We need to `set_balance` in benchmarking, so `Mutate` is needed

Reply:
It is possible to set custom where bounds for benchmarks, [like this](https://github.com/paritytech/polkadot-sdk/blob/8f6b6322c21c1eff2d075cd2f2af77191652df6f/polkadot/runtime/common/src/paras_registrar/mod.rs#L1507).  
So it should be possible to add a `<T as Config>::Currency: Mutate` there.